pkg/controller.v1beta1/experiment/util/status_util.go

```diff
@@ -192,7 +202,9 @@ func UpdateExperimentStatusCondition(collector *ExperimentsCollector, instance *
 	}
 
 	// First check if MaxFailedTrialCount is reached.
-	if (instance.Spec.MaxFailedTrialCount != nil) && (failedTrialsCount > *instance.Spec.MaxFailedTrialCount) {
+	if (instance.Spec.MaxFailedTrialCount != nil) &&
+		((*instance.Spec.MaxFailedTrialCount != *instance.Spec.MaxTrialCount && failedTrialsCount > *instance.Spec.MaxFailedTrialCount) ||
```

Comment:
Can you explain this condition from the current one?

Reply:
@johnugeorge  Thanks for your review!
Currently, if `*instance.Spec.MaxFailedTrialCount` is equal to `*instance.Spec.MaxTrialCount` and `failedTrialsCount` is equal to `*instance.Spec.MaxFailedTrialCount`, katib skip this condition and gives `Completed` to the Experiment status.

In other words, katib gives Experiment status `Completed` even though the number of failed Trials reaches `*instance.Spec.MaxFailedTrialCount`.

In this implementation, when the same condition as mentioned above, katib doesn't skip this condition and gives `Failed` to the experiment status.

Reply:
How does it matter with `*instance.Spec.MaxTrialCount`?  If I understand correctly, status is not set correctly when the number of failed Trials reaches `*instance.Spec.MaxFailedTrialCount`. Then, Isn't condition `failedTrialsCount >= *instance.Spec.MaxFailedTrialCount` enough ?

Reply:
In the current implementation, katib create `*instance.Spec.MaxFailedTrialCount + 1` times Trials if all Trials failed.
I implemented this part to keep this specification.
Should I change to `failedTrialsCount >= *instance.Spec.MaxFailedTrialCount`?

Reply:
Isn't failedTrialsCount >= *instance.Spec.MaxFailedTrialCount the right one? Experiment should be completed when failed trials reach MaxFailedTrialCount.

Reply:
> Isn't failedTrialsCount >= *instance.Spec.MaxFailedTrialCount the right one? Experiment should be completed when failed trials reach MaxFailedTrialCount.

Yes, I agree with @johnugeorge .
However, because as you say, there is an inconsistency between `maxFailedTrialCount` and `maxTrialCount`, we need to change definition of maxFailedTrialCount in order to change to `failedTrialsCount >= *instance.Spec.MaxFailedTrialCount` in this part.

Reply:
@tenzen-y Please go ahead and make this change - `failedTrialsCount >= *instance.Spec.MaxFailedTrialCount`

Reply:
@johnugeorge Sure.

Reply:
Done.